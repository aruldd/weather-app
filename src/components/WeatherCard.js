import { Card, Divider } from '@blueprintjs/core';
import PropTypes from 'prop-types';
import React from 'react';
import { getFormattedDateFromTS } from '../utils/dates';
import { UOM } from '../utils/units';
import { weatherProp } from '../types';
export const WeatherCard = (props) => {
  const { weather, system } = props;
  if (!weather) {
    return null;
  }
  const {
    dt,
    main,
    wind,
    visibility,
    weather: weather_,
    clouds,
    pop,
  } = weather;

  const weatherDescription = weather_[0];
  const { temp, feels_like, temp_min, temp_max, pressure, humidity } = main;
  const { speed, deg } = wind;
  const { all } = clouds;
  return (
    <Card className="weather-card">
      <div className="image">
        {' '}
        <img
          src={`https://openweathermap.org/img/wn/${weatherDescription.icon}@2x.png`}
        />
      </div>
      <div className="cell">
        <h2 className="time ">{getFormattedDateFromTS(dt, 'LT')}</h2>
        <h4 className="temp">
          {temp} {UOM['temp'][system]}
        </h4>
        <div className="description">{weatherDescription.description}</div>
      </div>
      <Divider />
      <div className="cell">
        <h2>Feels like</h2>
        <div className="value">
          {feels_like} {UOM['feels_like'][system]}
        </div>
      </div>
      <Divider />

      <div className="cell">
        <h2>Min/Max</h2>
        <div className="value">
          {temp_min} {UOM['temp_min'][system]}/{temp_max}{' '}
          {UOM['temp_max'][system]}
        </div>
      </div>
      <Divider />

      <div className="cell">
        <h2>Precipitation Probability</h2>
        <div className="value">
          {Math.round(pop * 100)} {UOM['pop'][system]}
        </div>
      </div>
      <Divider />

      <div className="cell">
        <h2>Pressure</h2>
        <div className="value">
          {pressure} {UOM['pressure'][system]}
        </div>
      </div>
      <Divider />

      <div className="cell">
        <h2>Humidity</h2>
        <div className="value">
          {humidity} {UOM['humidity'][system]}
        </div>
      </div>
      <Divider />

      <div className="cell">
        <h2>Wind</h2>
        <div className="value">
          Speed: {speed} {UOM['speed'][system]}
        </div>
        <div className="value">
          Direction: {deg} {UOM['deg'][system]}
        </div>
      </div>
      <Divider />

      <div className="cell">
        <h2>Cloudiness</h2>
        <div className="value">
          {all} {UOM['all'][system]}
        </div>
      </div>
      <Divider />

      <div className="cell">
        <h2>Visibility</h2>
        <div className="value">
          {visibility / 1000} {UOM['visibility'][system]}
        </div>
      </div>
    </Card>
  );
};

WeatherCard.propTypes = {
  weather: weatherProp,
  system: PropTypes.string.isRequired,
};
export default WeatherCard;
