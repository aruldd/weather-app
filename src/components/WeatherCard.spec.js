import React from 'react';
import { shallow, mount } from 'enzyme';
import WeatherCard from './WeatherCard';
import { create } from 'react-test-renderer';

describe('<WeatherCard />', () => {
  const weather = {
    dt: 1606284000,
    main: {
      temp: 25.73,
      feels_like: 24.6,
      temp_min: 25.45,
      temp_max: 25.73,
      pressure: 1008,
      sea_level: 1008,
      grnd_level: 1005,
      humidity: 92,
      temp_kf: 0.28,
    },
    weather: [
      {
        id: 502,
        main: 'Rain',
        description: 'heavy intensity rain',
        icon: '10d',
      },
    ],
    clouds: {
      all: 95,
    },
    wind: {
      speed: 10.2,
      deg: 30,
    },
    visibility: 10000,
    pop: 1,
    rain: {
      '3h': 16.31,
    },
    sys: {
      pod: 'd',
    },
    dt_txt: '2020-11-25 06:00:00',
  };

  it('Should render everything with imperial units', () => {
    const system = 'imperial';
    const component = create(<WeatherCard weather={weather} system={system} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render everything with metric units', () => {
    const system = 'metric';
    const component = create(<WeatherCard weather={weather} system={system} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
