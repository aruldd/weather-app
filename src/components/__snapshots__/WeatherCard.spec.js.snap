// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherCard /> Should render everything with imperial units 1`] = `
<div
  className="bp3-card bp3-elevation-0 weather-card"
>
  <div
    className="image"
  >
     
    <img
      src="https://openweathermap.org/img/wn/10d@2x.png"
    />
  </div>
  <div
    className="cell"
  >
    <h2
      className="time "
    >
      11:30 AM
    </h2>
    <h4
      className="temp"
    >
      25.73
       
      °F
    </h4>
    <div
      className="description"
    >
      heavy intensity rain
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Feels like
    </h2>
    <div
      className="value"
    >
      24.6
       
      °F
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Min/Max
    </h2>
    <div
      className="value"
    >
      25.45
       
      °F
      /
      25.73
       
      °F
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Precipitation Probability
    </h2>
    <div
      className="value"
    >
      100
       
      %
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Pressure
    </h2>
    <div
      className="value"
    >
      1008
       
      hPa
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Humidity
    </h2>
    <div
      className="value"
    >
      92
       
      %
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Wind
    </h2>
    <div
      className="value"
    >
      Speed: 
      10.2
       
      mph
    </div>
    <div
      className="value"
    >
      Direction: 
      30
       
      °
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Cloudiness
    </h2>
    <div
      className="value"
    >
      95
       
      %
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Visibility
    </h2>
    <div
      className="value"
    >
      10
       
      km
    </div>
  </div>
</div>
`;

exports[`<WeatherCard /> Should render everything with metric units 1`] = `
<div
  className="bp3-card bp3-elevation-0 weather-card"
>
  <div
    className="image"
  >
     
    <img
      src="https://openweathermap.org/img/wn/10d@2x.png"
    />
  </div>
  <div
    className="cell"
  >
    <h2
      className="time "
    >
      11:30 AM
    </h2>
    <h4
      className="temp"
    >
      25.73
       
      °C
    </h4>
    <div
      className="description"
    >
      heavy intensity rain
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Feels like
    </h2>
    <div
      className="value"
    >
      24.6
       
      °C
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Min/Max
    </h2>
    <div
      className="value"
    >
      25.45
       
      °C
      /
      25.73
       
      °C
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Precipitation Probability
    </h2>
    <div
      className="value"
    >
      100
       
      %
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Pressure
    </h2>
    <div
      className="value"
    >
      1008
       
      hPa
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Humidity
    </h2>
    <div
      className="value"
    >
      92
       
      %
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Wind
    </h2>
    <div
      className="value"
    >
      Speed: 
      10.2
       
      m/s
    </div>
    <div
      className="value"
    >
      Direction: 
      30
       
      °
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Cloudiness
    </h2>
    <div
      className="value"
    >
      95
       
      %
    </div>
  </div>
  <div
    className="bp3-divider"
  />
  <div
    className="cell"
  >
    <h2>
      Visibility
    </h2>
    <div
      className="value"
    >
      10
       
      km
    </div>
  </div>
</div>
`;
